Flow Aggregator panics when ClickHouse BeginTx fails

**Describe the bug**

When Flow Aggregator tries to export flow records to ClickHouse and the database connection fails at transaction start, the export path calls `Rollback()` on a nil transaction. That triggers a nil pointer panic and can crash the Flow Aggregator pod instead of returning the error and retrying on the next commit cycle.

This shows up when ClickHouse is down, misconfigured, or unreachable over the network.

**To Reproduce**

1. Enable ClickHouse export in the Flow Aggregator config.
2. Run Flow Aggregator with flow records queued for export.
3. Make `BeginTx` fail. For example, stop ClickHouse, point `clickHouse.databaseURL` at an unreachable host, or block network access to ClickHouse.
4. Wait for the next batch commit.

In unit tests, the same failure can be reproduced with sqlmock:

```go
mock.ExpectBegin().WillReturnError(fmt.Errorf("connection refused"))
chExportProc.batchCommitAll(ctx)
```

**Expected**

If `BeginTx` or `PrepareContext` fails, Flow Aggregator should log the error, return it from `batchCommitAll`, and keep running. Records should stay in the queue for a later retry. No panic.

**Actual behavior**

When `BeginTx` fails, `tx` is nil but `batchCommitAll` still calls `tx.Rollback()`. That causes a nil pointer dereference and the process panics.

Example from the failing test:

```
E0723 19:44:46.945292 clickhouseclient.go:266] "Error when preparing insert statement" err="connection refused"
panic: runtime error: invalid memory address or nil pointer dereference
database/sql.(*Tx).Rollback(...)
pkg/flowaggregator/clickhouseclient/clickhouseclient.go:267
```

**Versions:**

- Antrea version: (fill in the version or image tag you tested)
- Kubernetes version: (fill in)
- Container runtime: (fill in)
- Linux kernel version: (fill in)

**Additional context**

Relevant code in `pkg/flowaggregator/clickhouseclient/clickhouseclient.go`:

```go
tx, err := ch.db.BeginTx(ctx, nil)
if err == nil {
    stmt, err = tx.PrepareContext(ctx, insertQuery)
}
if err != nil {
    klog.ErrorS(err, "Error when preparing insert statement")
    _ = tx.Rollback() // tx is nil when BeginTx fails
    return 0, err
}
```

Suggested fix:

```go
if tx != nil {
    _ = tx.Rollback()
}
```

A regression test `TestBatchCommitAllBeginTxError` covers the `BeginTx` failure path.

--- a/pkg/flowaggregator/clickhouseclient/clickhouseclient.go
+++ b/pkg/flowaggregator/clickhouseclient/clickhouseclient.go
@@ -264,7 +264,9 @@ func (ch *ClickHouseExportProcess) batchCommitAll(ctx context.Context) (int, err
 	}
 	if err != nil {
 		klog.ErrorS(err, "Error when preparing insert statement")
-		_ = tx.Rollback()
+		if tx != nil {
+			_ = tx.Rollback()
+		}
 		return 0, err
 	}
 

--- a/pkg/flowaggregator/clickhouseclient/clickhouseclient_test.go
+++ b/pkg/flowaggregator/clickhouseclient/clickhouseclient_test.go
@@ -192,6 +192,27 @@ func TestBatchCommitAllError(t *testing.T) {
 	assert.NoError(t, mock.ExpectationsWereMet(), "unfulfilled expectations for db sql operation")
 }
 
+func TestBatchCommitAllBeginError(t *testing.T) {
+	db, mock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+	require.NoError(t, err, "error when opening a stub database connection")
+	defer db.Close()
+
+	chExportProc := &ClickHouseExportProcess{
+		db:        db,
+		queueSize: maxQueueSize,
+	}
+	recordRow := flowrecord.FlowRecord{}
+	chExportProc.deque.PushBack(&recordRow)
+
+	mock.ExpectBegin().WillReturnError(fmt.Errorf("mock error for begin transaction"))
+
+	count, err := chExportProc.batchCommitAll(t.Context())
+	assert.Error(t, err, "expected error when SQL transaction cannot begin")
+	assert.Equal(t, 0, count)
+	assert.Equal(t, 1, chExportProc.deque.Len())
+	assert.NoError(t, mock.ExpectationsWereMet(), "unfulfilled expectations for db sql operation")
+}
+
 func TestPushRecordsToFrontOfQueue(t *testing.T) {
 	chExportProc := &ClickHouseExportProcess{
 		queueSize: 4,

